박정근/BOJ/Week27/[10423]전기가_부족해.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Author    : Park Jeong Geun
+ * Date      : 2024.12.2(Mon)
+ * Runtime   : 712ms
+ * Memory    : 66424KB
+ * Algorithm : mst
+ */
+
+// >> 첫 번째 풀이 ( 최소 스패닝 트리 (크루스칼) )
+// 최소 스패닝 트리로 풀이했습니다. 크루스칼 알고리즘을 사용했습니다. (Solve Time : 0h 43m)
+// Edge 클래스와 MinnimumSpanningTree 클래스를 만들어, 제 기능을 잘 수행하도록 캡슐화하는 데에 집중했습니다.
+// 일반적인 최소 스패닝 트리 문제와 같지만, 이으려는 두 그룹 다 발전소와 이미 이어진 그룹이라면 이을 필요가 없으므로 다음 간선을 이으면 됩니다.
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.StringTokenizer;
+import java.util.Arrays;
+import java.util.Stack;
+
+class Edge {
+    int u, v, w;
+    public Edge(int u, int v, int w) {
+        this.u = u;
+        this.v = v;
+        this.w = w;
+    }
+}
+
+class MinnimumSpanningTree {
+    private int n, m;
+    private int[] power_station; // 발전소 여부
+    private Edge[] edges; // 간선 배열
+    private int[] parent; // 부모 노드 배열
+    
+    public MinnimumSpanningTree(int n, int m) {
+        this.n = n;
+        this.m = m;
+        
+        edges = new Edge[m];
+
+        power_station = new int[n+1];
+        Arrays.fill(power_station, 0);
+
+        parent = new int[n+1];
+        for (int i = 1; i <= n; i++) parent[i] = i;
+    }
+
+    public void addPowerStation(int p) { power_station[p] = 1; }
+    public void addEdge(int i, int u, int v, int w) { edges[i] = new Edge(u, v, w); }
+    public void sortEdges() { 
+        // Kruskal : 간선 정렬
+        Arrays.sort(edges, (e1, e2) -> Integer.compare(e1.w, e2.w)); 
+    }
+    public boolean isPower(int a) { 
+        // power_station[a] == 1 쓰기가 귀찮아서..
+        if (power_station[a] == 1) return true;
+        return false;
+    }
+
+    public int find(int a) {
+        // Union Find : 부모 노드를 스택으로 찾으며, 나머지 노드 모두 부모를 동기화하는 함수
+        Stack<Integer> citys = new Stack<>();
+
+        while (a != parent[a]) {
+            citys.push(a);
+            a = parent[a];
+        }
+
+        while (!citys.isEmpty()) {
+            parent[citys.peek()] = a; 
+            citys.pop();
```

Comment:
```suggestion
            parent[citys.pop()] = a; 
```

peek() 이랑 pop() 나누지 않아도 될 것 같습니다.

Reply:
감사합니다!